ui/src/shared/components/TableGraph.js

```diff
@@ -160,6 +177,15 @@ TableGraph.propTypes = {
   tableOptions: shape({}),
   hoverTime: string,
   onSetHoverTime: func,
+  colors: arrayOf(
+    shape({
+      type: string.isRequired,
+      hex: string.isRequired,
+      id: string.isRequired,
+      name: string.isRequired,
+      value: string.isRequired,
+    }).isRequired
+  ),
 }
 
 export default TableGraph
```

Comment:
OK. this one is important. for MultiGrid to get the memo that one of the props that cellrenderer uses has changed, (such as when you add a new threshold color), it needs to receive it as a prop. When one of Multi-Grid's props changes, it goes ahead and re-runs cell renderer and applies the new colors. This way the table will update live in CEO view. you just need to add `colors={colors}` to line 166 and get colors from props too.

Reply:
ahhhhh that explains everything